lib/media/streaming_engine.js

```diff
@@ -1586,34 +1589,58 @@ shaka.media.StreamingEngine = class {
         reference.startTime <= appendWindowEnd,
         logPrefix + ' segment should start before append window end');
 
+    const codecs = MimeUtils.getCodecBase(mediaState.stream.codecs);
     const timestampOffset = reference.timestampOffset;
+
     if (timestampOffset != mediaState.lastTimestampOffset ||
         appendWindowStart != mediaState.lastAppendWindowStart ||
-        appendWindowEnd != mediaState.lastAppendWindowEnd) {
+        appendWindowEnd != mediaState.lastAppendWindowEnd ||
+        codecs != mediaState.lastCodecs) {
       shaka.log.v1(logPrefix, 'setting timestamp offset to ' + timestampOffset);
       shaka.log.v1(logPrefix,
           'setting append window start to ' + appendWindowStart);
       shaka.log.v1(logPrefix,
           'setting append window end to ' + appendWindowEnd);
 
+      if (codecs != mediaState.lastCodecs) {
+        if (mediaState.type === ContentType.VIDEO) {
+          /** @type {shaka.media.StreamingEngine.MediaState_} */
+          const audioState = this.mediaStates_.get(ContentType.AUDIO);
+          if (audioState) {
+            audioState.lastInitSegmentReference = null;
+            this.abortOperations_(audioState);
+          }
+        } else if (mediaState.type === ContentType.AUDIO) {
+          /** @type {shaka.media.StreamingEngine.MediaState_} */
+          const videoState = this.mediaStates_.get(ContentType.VIDEO);
+          if (videoState) {
+            videoState.lastInitSegmentReference = null;
+            this.abortOperations_(videoState);
+          }
+        }
+      }
+
       const setProperties = async () => {
         try {
+          mediaState.lastCodecs = codecs;
           mediaState.lastAppendWindowStart = appendWindowStart;
           mediaState.lastAppendWindowEnd = appendWindowEnd;
           mediaState.lastTimestampOffset = timestampOffset;
 
           await this.playerInterface_.mediaSourceEngine.setStreamProperties(
               mediaState.type, timestampOffset, appendWindowStart,
-              appendWindowEnd, this.manifest_.sequenceMode);
+              appendWindowEnd, this.manifest_.sequenceMode,
+              mediaState.stream);
         } catch (error) {
           mediaState.lastAppendWindowStart = null;
           mediaState.lastAppendWindowEnd = null;
           mediaState.lastTimestampOffset = null;
+          mediaState.lastCodecs = null;
 
           throw error;
         }
       };
-      operations.push(setProperties());
+      await setProperties();
```

Comment:
Why did you change this?  With this change, we will strictly wait for setStreamProperties() to complete before beginning to fetch the init segment.  I don't see any reason those can't happen in parallel.